src/elastio-snap.c

```diff
@@ -626,6 +626,11 @@ static inline MRF_RETURN_TYPE elastio_snap_null_mrf(struct request_queue *q, str
 	percpu_ref_get(&q->q_usage_counter);
 	return blk_mq_make_request(q, bio);
 }
+#elif defined HAVE_BLK_MQ_MAKE_REQUEST
+static inline MRF_RETURN_TYPE elastio_snap_null_mrf(struct request_queue *q, struct bio *bio){
+	smp_wmb();
+	return blk_mq_make_request(q, bio);    
+}
```

Comment:
The tests are failing on the loopback device after the snapshot destroy operation with this dmesg output:
```
[  +0.000001] elastio-snap: received destroy ioctl - 14
[  +0.000000] elastio-snap: device specified does not exist: -2
[  +0.000414] elastio-snap: error during destroy ioctl handler: -2
[  +0.000408] elastio-snap: minor range = 23 - 0
[  +0.072161] ------------[ cut here ]------------
[  +0.000039] percpu ref (blk_queue_usage_counter_release) <= 0 (-149) after switching to atomic
[  +0.000042] WARNING: CPU: 1 PID: 0 at lib/percpu-refcount.c:161 percpu_ref_switch_to_atomic_rcu+0x135/0x140
[  +0.000002] Modules linked in: elastio_snap(OE) loop ipt_MASQUERADE nf_conntrack_netlink nft_counter xt_addrtype nft_compat br_netfilter bridge stp llc nft_masq nft_fib_inet nft_fib_ipv4 nft_fib_ipv6 nft_fib >
[  +0.000034] CPU: 1 PID: 0 Comm: swapper/1 Kdump: loaded Tainted: G           OE    --------- -  - 4.18.0-305.3.1.el8.x86_64 #1
[  +0.000001] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 1.13.0-1ubuntu1.1 04/01/2014
[  +0.000002] RIP: 0010:percpu_ref_switch_to_atomic_rcu+0x135/0x140
[  +0.000002] Code: 80 3d 06 ac 34 01 00 0f 85 53 ff ff ff 49 8b 54 24 d8 49 8b 74 24 e8 48 c7 c7 d8 56 90 93 c6 05 e8 ab 34 01 01 e8 74 9a c5 ff <0f> 0b e9 2f ff ff ff 0f 1f 40 00 41 54 49 89 f4 55 53 48 89 fb>
[  +0.000001] RSP: 0018:ffffaf06c0d14eb8 EFLAGS: 00010286
[  +0.000001] RAX: 0000000000000000 RBX: 7fffffffffffff6a RCX: 0000000000000000
[  +0.000001] RDX: 0000000000000002 RSI: ffffffff946b8f32 RDI: 0000000000000246
[  +0.000001] RBP: 000030a7c8001c00 R08: ffffffff946b8ee0 R09: 0000000000000052
[  +0.000000] R10: ffffffff946b92c0 R11: 0000000080000001 R12: ffff9e5ebd1f61a0
[  +0.000001] R13: ffffffff9401ff00 R14: ffffffff9295e80f R15: ffff9e5ef7caacd0
[  +0.000004] FS:  0000000000000000(0000) GS:ffff9e5ef7c80000(0000) knlGS:0000000000000000
[  +0.000017] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[  +0.000001] CR2: 00007fda5b32eb40 CR3: 00000001f24ca000 CR4: 0000000000350ee0
[  +0.000003] Call Trace:
[  +0.000015]  <IRQ>
[  +0.000019]  rcu_do_batch+0x1a4/0x3d0
[  +0.000014]  rcu_core+0x18f/0x2f0
[  +0.000012]  __do_softirq+0xd7/0x2d6
[  +0.000020]  irq_exit+0xf7/0x100
[  +0.000008]  smp_apic_timer_interrupt+0x74/0x130
[  +0.000006]  apic_timer_interrupt+0xf/0x20
[  +0.000001]  </IRQ>
[  +0.000002] RIP: 0010:native_safe_halt+0xe/0x10
[  +0.000002] Code: ff ff 7f c3 65 48 8b 04 25 40 5c 01 00 f0 80 48 02 20 48 8b 00 a8 08 75 c4 eb 80 90 e9 07 00 00 00 0f 00 2d 06 d4 4b 00 fb f4 <c3> 90 e9 07 00 00 00 0f 00 2d f6 d3 4b 00 f4 c3 90 90 0f 1f 44>
[  +0.000000] RSP: 0018:ffffaf06c0cc7ea0 EFLAGS: 00000212 ORIG_RAX: ffffffffffffff13
[  +0.000002] RAX: ffffffff9314c310 RBX: 0000000000000001 RCX: ffff9e5ee5282db8
[  +0.000000] RDX: 0000000000000001 RSI: 0000000000000000 RDI: 00004b552d298880
[  +0.000001] RBP: 0000000000000001 R08: 0000748c2e146d4e R09: ffff9e5dd6e6c7c0
[  +0.000000] R10: 0000000000000018 R11: 0000000000000001 R12: ffffffffffffffff
[  +0.000001] R13: 0000000000000000 R14: 0000000000000000 R15: ffff9e5dc6e15f00
[  +0.000002]  ? __sched_text_end+0x7/0x7
[  +0.000007]  default_idle+0xa/0x10
[  +0.000002]  default_idle_call+0x40/0xf0
[  +0.000003]  do_idle+0x1f4/0x260
[  +0.000008]  cpu_startup_entry+0x6f/0x80
[  +0.000007]  start_secondary+0x199/0x1e0
[  +0.000010]  secondary_startup_64_no_verify+0xc2/0xcb
[  +0.000008] ---[ end trace 0f4492ee80d9bec3 ]---
```
So, you should reuse already existing function `elastio_snap_null_mrf` few lines above
https://github.com/elastio/elastio-snap/blob/162d2640ae468ccaee43481a837504930ead5c0e/src/elastio-snap.c#L625-L628
with the `percpu_ref_get(&q->q_usage_counter);` instead of the `smp_wmb();`. It works fine on the kernel 5.8 and will work with the 4.18-0.305.